src/modules/cbm/geometry/Parser.h

```diff
@@ -32,23 +33,31 @@ class Parser {
   const unsigned int MaxFENId{23};
   const unsigned int MaxReadoutsInPacket{600};
 
-// From TTLMon ICD version 1 draft 2
+// From TTLMon ICD (CBM ICD) version 1 draft 2 - 4
 // Preliminary agreed 2023 09 12 (Francesco, Farnaz, Fabio)
 #define DATASIZE 20
-  struct Data {
+
+  struct Position {
+    uint16_t XPos;
+    uint16_t YPos;
+  } __attribute__((packed));
+
+  struct CbmReadout {
     uint8_t FiberId;
     uint8_t FENId;
     uint16_t DataLength;
     uint32_t TimeHigh;
     uint32_t TimeLow;
-    uint8_t Pos;
-    uint8_t Channel;
+    uint8_t Type;
     uint16_t ADC;
-    uint16_t XPos;
-    uint16_t YPos;
+    uint8_t Channel;
+    union {
+      Position Pos;
```

Comment:
If you don't plan to use the struct Position elsewhere you could put it into the union